### DISCUSSION TITLE
Docker : add a "wait for the db" feature

### DISCUSSION THREAD

**Discussion by romu70** [go-shiori/shiori#587] [Feature requests]
Hi all.
Currently the Shiori container doesn't start if the DB is not reachable. It would be cool to add a feature which simply waits for the DB availability on a regular basis. This way, we wouldn't have to deal with containers start sequence.

**Comment by fmartingr** [comment]
I will make a mental note for this, would be interesting to have for a `liveness` endpoint also. In the meanwhile, you can use a `wait-for-it`[^1} approach in docker, either you a new container wait for the DB, and the application container wait for this container, or a custom container with this command before the serve command. I believe there's a `wait-for-db` somewhere too.

[^1]: https://github.com/vishnubob/wait-for-it

**Reply by fmartingr** [reply]
I've made a proper issue for this: https://github.com/go-shiori/shiori/issues/588

**Reply by romu70** [reply]
Thx a lot
